opentelekomcloud/resource_opentelekomcloud_rds_instance_v1.go

```diff
@@ -0,0 +1,618 @@
+	//		"[DEBUG] Error saving haSet to Rds instance (%s): %s", d.Id(), err)
+	//}
+
+	d.Set("updated", instance.Updated)
+	d.Set("created", instance.Created)
+	return nil
+}
+
+func resourceInstanceDelete(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	client, err := config.RdsV1Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("Error creating OpenTelekomCloud rds client: %s ", err)
+	}
+
+	log.Printf("[DEBUG] Deleting Instance %s", d.Id())
+
+	id := d.Id()
+	result := instances.Delete(client, id)
+	if result.Err != nil {
+		return err
+	}
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"ACTIVE"},
+		Target:     []string{"DELETED"},
+		Refresh:    InstanceStateRefreshFunc(client, id),
+		Timeout:    d.Timeout(schema.TimeoutCreate),
+		Delay:      15 * time.Second,
+		MinTimeout: 3 * time.Second,
+	}
+
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf(
+			"Error waiting for instance (%s) to be deleted: %s ",
+			id, err)
+	}
+	time.Sleep(80 * time.Second)
+	log.Printf("[DEBUG] Successfully deleted instance %s", id)
+	return nil
+}
+
+func InstanceStateUpdateRefreshFunc(client *golangsdk.ServiceClient, instanceID string, size int) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		instance, err := instances.Get(client, instanceID).Extract()
+		if err != nil {
+			if _, ok := err.(golangsdk.ErrDefault404); ok {
+				return instance, "DELETED", nil
+			}
+			return nil, "", err
+		}
+		log.Printf("[DEBUG] Updating instance.Volume : %+v", instance.Volume)
+		if instance.Volume.Size == size {
+			return instance, "UPDATED", nil
+		}
+
+		return instance, instance.Status, nil
+	}
+}
+
+func InstanceStateFlavorUpdateRefreshFunc(client *golangsdk.ServiceClient, instanceID string, flavorID string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		instance, err := instances.Get(client, instanceID).Extract()
+		if err != nil {
+			if _, ok := err.(golangsdk.ErrDefault404); ok {
+				return instance, "DELETED", nil
+			}
+			return nil, "", err
+		}
+		//log.Printf("[DEBUG] Updating instance.Flavor : %+v", instance.Flavor)
+		//if instance.Flavor.Id == flavorID {
+		//    return instance, "FLAVOR_UPDATED", nil
+		//}
+
+		return instance, instance.Status, nil
+	}
+}
+
+func resourceInstanceUpdate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	client, err := config.RdsV1Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("Error Updating OpenTelekomCloud rds client: %s ", err)
+	}
+
+	log.Printf("[DEBUG] Updating instances %s", d.Id())
+	id := d.Id()
+
+	if d.HasChange("volume") {
+		var updateOpts instances.UpdateOps
+		volume := make(map[string]interface{})
+		volumeRaw := d.Get("volume").([]interface{})
+		log.Printf("[DEBUG] volumeRaw: %+v", volumeRaw)
+		if len(volumeRaw) == 1 {
+			volume["size"] = volumeRaw[0].(map[string]interface{})["size"].(int)
```

Comment:
This is a potentially fragile way to do a type assertion (is it possible for the value to be `nil`)? May need to use the version that returns `ok` but I'll leave that up to you. As it stands this could potentially cause a panic.